### DISCUSSION TITLE
Difficulty in calculating the correct position in the webots environment

### DISCUSSION THREAD

**Discussion by 233Xiaowu** [cyberbotics/webots#6547] [Q&A]
**Describe the Bug**
Thank you very much for your work, at the moment I would like to be able to reproduce the ORBSLAM3 program and the comparison of the trajectory accuracy between the slam program and the trajectory truth in the simulation environment you have provided, but at the moment there are a few issues that arise.

**Steps to Reproduce**
1. I have calibrated well in the webots environment that you provide
2. I implemented the extraction of the truth value of the robot body trajectory via pycharm
3. I use ros to pass the image information into the SLAM program and calculate the trajectory
4. I ended up using evo to achieve accuracy comparisons between trajectories

**Expected behavior**
What I'm hoping for is that the trajectory truth should not be too far off from the historical trajectory computed by the SLAM program without influence

**Screenshots**
Here's a scree
![QQ20240508-154024](https://github.com/cyberbotics/webots/assets/116873301/9aa49e23-1671-4648-b20d-8564e1facbd4)
nshot of my calibration

Here's a reproduction of my ORBSLAM3 program in webots
![QQ20240508-154129](https://github.com/cyberbotics/webots/assets/116873301/ea9f4c36-dbea-406e-a3b9-0c1bc9d6768f)

Here's my final trajectory comparison Where the gray line segments are the true values of the trajectories extracted from the webots, and the colored line segments are the trajectories calculated by ORBSLAM3
![image](https://github.com/cyberbotics/webots/assets/116873301/837126ee-d717-4b0f-bdcb-fee085d7aa9a)

**System**
webots R2023A
ubuntu 20.04
ORBSLAM3 
evo

**Comment by AntonioLuisGH** [comment]
Hey, this looks like a really cool setup! Just a fellow user chiming in, but I've run into similar trajectory mismatches before when mixing simulators and SLAM.

A couple of things that might be causing this:

Coordinate Frame Mismatch: Webots, ROS, and camera sensors often use completely different axis conventions (e.g., Z-up for the world vs. Z-forward/Y-down for the camera). If the transformations aren't perfectly mapped, the trajectories will look way off.

Origin and Scale: SLAM usually starts at its own arbitrary local origin (0,0,0), whereas your Webots ground truth is based on the global world coordinates. Also, if you are running monocular ORBSLAM3, you might be hitting scale ambiguity.

When you run evo to compare them, are you using the alignment flags? You usually have to mathematically align the SLAM trajectory to the ground truth. Try running your evo command with -a (align origin/rotation) and -s (correct scale) and see if they snap together!
